### PR TITLE
Well group override in upload form for Pulse3d 0.30.3 and above

### PR DESCRIPTION
**Pull request by luciipak**

### CODE REVIEW

File: jobs/pulse3d/pulse3d-worker/src/main.py [hunk 1]
```diff
@@ -208,7 +209,23 @@ async def process(con, item):
...
+
+                            indices_to_keep = indices_after_start & indices_before_end
+                            windowed_well_data = interpolated_well_data[:, indices_to_keep]
+
+                            peaks, valleys = peak_detector(windowed_well_data, **peak_detector_args)
+                            idx_shift = np.argmax(indices_after_start)
+                            peaks += idx_shift
+                            valleys += idx_shift
+                        except:
+                            peaks, valleys = peak_detector(interpolated_well_data, **peak_detector_args)
```
Comment (luciipak) [inline jobs/pulse3d/pulse3d-worker/src/main.py:228]: @tannermpeterson just want to double check this is supposed to be here?

**Comment by luciipak** [comment]
This has a bunch of commits because I merged in all changes from 02-16-23 into 03-02-23 and merged those into this branch.

File: jobs/pulse3d/pulse3d-worker/src/main.py (hunk 1 above)
Comment (tannermpeterson) [reply jobs/pulse3d/pulse3d-worker/src/main.py:227]: No this whole try/except isn't necessary anymore, should just be `peaks, valleys = peak_detector(interpolated_well_data, **peak_detector_args)`

File: deployments/pulse3d/services/pulse3d/src/core/config.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 config = Config(".env")
 # version used to tag the docker image
-VERSION = "0.6.1"
+VERSION = "0.6.2"
```
Comment (tannermpeterson) [inline deployments/pulse3d/services/pulse3d/src/core/config.py:6]: Does the makefile match this version?

Comment (luciipak) [reply deployments/pulse3d/services/pulse3d/src/core/config.py:6]: Nikita actually just bumped these anyway, so I'll increase everything

File: frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js [hunk 3]
```diff
@@ -140,37 +146,47 @@ const SmallLabel = styled.label`
...
 `;
 
+const AdvAnalysisContainer = styled.div`
+display: flex;
+flex-direction row;
+height: 157px;
+`;
+
 export default function AnalysisParamForm({
-  inputVals,
```
Comment (tannermpeterson) [inline frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js:144]: Why was this removed?

Comment (luciipak) [reply frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js:144]: analysisParams were being passed in twice

File: frontend/pulse3d/src/pages/upload-form.js [hunk 4]
```diff
@@ -176,7 +178,7 @@ export default function UploadForm() {
         usageQuota.limits &&
         parseInt(usageQuota.limits.jobs) !== -1
     );
-  }, [paramErrors, files, inProgress]);
+  }, [paramErrors, files, inProgress, wellGroupErr]);
```
Comment (tannermpeterson) [inline frontend/pulse3d/src/pages/upload-form.js:181]: Why isn't `wellGroupErr` just another possible error under `paramErrors`?

Comment (luciipak) [reply frontend/pulse3d/src/pages/upload-form.js:181]: Because the structure is different and became overly complicated with handling errors for both the label name input and the well input. it would have required changes to how the current `paramErrors` get checked as opposed to checking this extra booleann value

Comment (luciipak) [reply frontend/pulse3d/src/pages/upload-form.js:181]: If you think it would be better in paramErrors, I can refactor to add it in

File: frontend/pulse3d/src/components/uploadForm/WellGroups.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+  const removeWellGroup = () => {
+    localGroups.pop();
+    setLocalGroups([...localGroups]);
+  };
+
+  return (
+    <>
+      <SectionLabel>
+        Well Groupings{" "}
```
[the remark this answers is not in the document]
Comment (luciipak) [reply frontend/pulse3d/src/components/uploadForm/WellGroups.js:163]: pre-commit does this

File: frontend/pulse3d/src/pages/upload-form.js (hunk 4 above)
Comment (tannermpeterson) [reply frontend/pulse3d/src/pages/upload-form.js:181]: Ideally we should have just one place to check for errors in the analysis form, but if it's going to take a significant refactor we can do it later
